src/btree/bt_cursor.c

```diff
@@ -802,9 +804,14 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
      * Check that the provided search key is within bounds. If not, return WT_NOTFOUND and early
      * exit.
      */
-    if (btree->type == BTREE_ROW && WT_CURSOR_BOUNDS_SET(cursor)) {
-        WT_ERR(
-          __btcur_bounds_contains_key(session, cursor, &cursor->key, &key_out_of_bounds, NULL));
+    if (WT_CURSOR_BOUNDS_SET(cursor)) {
+        if (btree->type == BTREE_ROW)
+            WT_ERR(__btcur_bounds_contains_key(
+              session, cursor, &cursor->key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
```

Comment:
I think this unintentionally supports BTREE_COL_FIX.

Reply:
I think so too